### PR TITLE
Move CodeQL from Mac to Mac Legacy

### PR DESCRIPTION
**Pull request by softins**
**Short description of changes**

Moved the CodeQL for Mac from the signed main build to the unsigned legacy build

CHANGELOG: Moved CodeQL from Mac to Mac Legacy to work around signing incompatibility.

**Context: Fixes an issue?**

Currently, CodeQL interferes with the signing of Mac builds. This separates the CodeQL change from #2563, leaving the latter just for the entitlements fix.

**Does this change need documentation? What needs to be documented and how?**

Probably not

**Status of this Pull Request**

Tested with a manual workflow run

**What is missing until this pull request can be merged?**

Ready to merge

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [ ] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [ ] I've filled all the content above

### CODE REVIEW

**Review by ann0see — APPROVED** [review]
Since it worked for me, I‘m happy to approve

File: .github/workflows/autobuild.yml [hunk 1]
```diff
@@ -122,19 +122,21 @@ jobs:
...
             run_codeql:             false
 
-          - config_name:            MacOS (artifacts+codeQL)
+          - config_name:            MacOS (artifacts)
             target_os:              macos
             # Stay on 10.15 as long as we use dmgbuild which does not work with 11's hdiutil (?):
             building_on_os:         macos-10.15
             base_command:           QT_VERSION=5.15.2 SIGN_IF_POSSIBLE=1 ./.github/autobuild/mac.sh
-            run_codeql:             true
+            # Disable CodeQL on mac as it interferes with signing the binaries
```
Comment (hoffie) [inline .github/workflows/autobuild.yml:130]:
```suggestion
            # Disable CodeQL on mac as it interferes with signing the binaries (signing hangs)
```

File: .github/workflows/autobuild.yml [hunk 2]
```diff
@@ -122,19 +122,21 @@ jobs:
...
+            run_codeql:             false
             xcode_version:          12.1.1
 
-          - config_name:            MacOS Legacy (artifacts)
+          - config_name:            MacOS Legacy (artifacts+CodeQL)
```
[the remark this answers is not in the document]
Comment (ann0see) [reply .github/workflows/autobuild.yml:134]: ... and link this PR.

**Comment by ann0see** [comment]
@softins can you add the suggestions + link to this PR too?

**Comment by softins** [comment]
> @softins can you add the suggestions + link to this PR too?

Sorry, I didn't notice them before merging (I did it on the iPad app, which seems to hide a lot!). Maybe I'll merge them directly.

**Pull request merged**